tests/python-gpu/test_from_cupy.py

```diff
@@ -220,4 +220,5 @@ def test_specified_device(self):
             np.float32, xgb.DeviceQuantileDMatrix, np.nan)
         with pytest.raises(xgb.core.XGBoostError):
             xgb.train({'tree_method': 'gpu_hist', 'gpu_id': 1},
+                      'fail_on_invalid_gpu_id': True},
```

Comment:
This should fail without additional parameters assuming there is more than 1 device available during testing.